src/components/QuizQuestion.vue

```diff
@@ -35,16 +35,22 @@ export default {
       type: Number,
       required: true,
     },
-    goToNextStep: {
-      type: Function,
-      required: true,
-    },
   },
 
   computed: {
-    answerType() {
+    answerComponent() {
       return this.question.type === 'text' ? QuizAnswerText : QuizAnswerSelect;
     },
   },
+
+  methods: {
+    submitAnswer(answer) {
+      if (answer) {
+        this.$emit('goToNextStep', {
```

Comment:
Hmm 🤔 it seems like this component's responsibility should be only submitting the answer, and what follows - should be the parent responsibility (going to the next step). Do you agree?

Reply:
Yeah, I think it's about naming. It could be: submitA nswer like in other places for consistency because this component shouldn't be aware of what happens next, it only should emit answer.